Issue finding zstd

The library worked on most of my machines, but strangely on one XUbuntu with version 6.18 httplib's CMakeLists could not find zstd, even though it was installed and `/usr/lib/x86_64-linux-gnu/cmake/zstd` contained `zstdConfig.cmake` and `zstdTargets.cmake`.

```
CMake Error at build/_deps/cpp-httplib-src/CMakeLists.txt:273 (target_link_libraries): The link interface of target "httplib" contains: zstd::libzstd but the target was not found. Possible reasons include: * There is a typo in the target name. * A find_package call is missing for an IMPORTED target. * An ALIAS target is missing.
```

I am unsure why exactly it failed on that particular machine and worked on my other.

However, i finally managed to make httplib still compile after adding:

```cmake
find_package(zstd CONFIG REQUIRED)
if (NOT TARGET zstd::libzstd)
  add_library(zstd::libzstd INTERFACE IMPORTED)
  set_property(TARGET zstd::libzstd PROPERTY
    INTERFACE_LINK_LIBRARIES zstd)
endif()
```

Hope this helps, maybe you can use that information to harden your library.

P.S.: Thanks for your work, really enjoy your library so far.

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -185,10 +185,11 @@ elseif(HTTPLIB_USE_BROTLI_IF_AVAILABLE)
 	set(HTTPLIB_IS_USING_BROTLI ${Brotli_FOUND})
 endif()
 
-# NOTE: When using cpp-httplib as a subproject (e.g., via FetchContent), the zstd::libzstd target may not be visible in the parent project scope.
-# If you encounter a "target not found" error, see https://github.com/yhirose/cpp-httplib/issues/2313 for a workaround.
+# NOTE:
+# zstd < 1.5.6 does not provide the CMake imported target `zstd::libzstd`.
+# Older versions must be consumed via their pkg-config file.
 if(HTTPLIB_REQUIRE_ZSTD)
-	find_package(zstd)
+	find_package(zstd 1.5.6 CONFIG)
 	if(NOT zstd_FOUND)
 		find_package(PkgConfig REQUIRED)
 		pkg_check_modules(zstd REQUIRED IMPORTED_TARGET libzstd)
@@ -196,7 +197,7 @@ if(HTTPLIB_REQUIRE_ZSTD)
 	endif()
 	set(HTTPLIB_IS_USING_ZSTD TRUE)
 elseif(HTTPLIB_USE_ZSTD_IF_AVAILABLE)
-	find_package(zstd QUIET)
+	find_package(zstd 1.5.6 CONFIG QUIET)
 	if(NOT zstd_FOUND)
 		find_package(PkgConfig QUIET)
 		if(PKG_CONFIG_FOUND)

